Error about updating tidb_ttl_table_status

## Bug Report

### 1. Minimal reproduce step (Required)

Create a table with a TTL

### 2. What did you expect to see? (Required)

No TTL related errors in the logs

### 3. What did you see instead (Required)

```
[2022/12/29 09:28:44.308 +01:00] [ERROR] [job_manager.go:413] ["change ttl job status"] [ttl-worker=manager] [jobID=ceae5034-8752-11ed-b1e3-5405db7aad56] [table=t1] [error="execute sql: UPDATE mysql.tidb_ttl_table_status SET current_job_status = %? WHERE table_id = %? AND current_job_status = %? AND current_job_id = %?: unsupported 1-th argument: running"] [id=ceae5034-8752-11ed-b1e3-5405db7aad56]
```

Same message, but formatted differently:
```
[2022/12/29 09:28:44.308 +01:00]
[ERROR]
[job_manager.go:413]
["change ttl job status"]
[ttl-worker=manager]
[jobID=ceae5034-8752-11ed-b1e3-5405db7aad56] [table=t1]
[error="execute sql: UPDATE mysql.tidb_ttl_table_status SET current_job_status = %? WHERE table_id = %? AND current_job_status = %? AND current_job_id = %?: unsupported 1-th argument: running"]
[id=ceae5034-8752-11ed-b1e3-5405db7aad56]
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->
```
sql> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v6.5.0
Edition: Community
Git Commit Hash: 706c3fa3c526cdba5b3e9f066b1a568fb96c56e3
Git Branch: heads/refs/tags/v6.5.0
UTC Build Time: 2022-12-27 03:50:44
GoVersion: go1.19.3
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv
1 row in set (0.0011 sec)
```

Same happens on master @ cf3494140

```
sql> set global tidb_ttl_job_run_interval="10m";
Query OK, 0 rows affected (0.0073 sec)

sql> CREATE TABLE `t1` (`id` int(11) NOT NULL, `created_at` timestamp(6) DEFAULT CURRENT_TIMESTAMP(6), PRIMARY KEY (`id`) CLUSTERED ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin TTL=`created_at` + INTERVAL 15 SECOND TTL_ENABLE='ON';
Query OK, 0 rows affected (0.0713 sec)

sql> INSERT INTO t1(id) VALUES (1);
Query OK, 1 row affected (0.0019 sec)

sql> TABLE t1;
+----+----------------------------+
| id | created_at                 |
+----+----------------------------+
|  1 | 2022-12-29 09:41:56.748816 |
+----+----------------------------+
1 row in set (0.0021 sec)
```

```
[2022/12/29 09:43:25.106 +01:00] [INFO] [job_manager.go:377] ["try lock new job"] [ttl-worker=manager] [tableID=80]
[2022/12/29 09:43:25.108 +01:00] [INFO] [job_manager.go:380] ["append new running job"] [ttl-worker=manager] [jobID=dbaf9b58-8754-11ed-9b49-5405db7aad56] [tableID=80]
[2022/12/29 09:43:25.108 +01:00] [ERROR] [job_manager.go:413] ["change ttl job status"] [ttl-worker=manager] [jobID=dbaf9b58-8754-11ed-9b49-5405db7aad56] [table=t1] [error="execute sql: UPDATE mysql.tidb_ttl_table_status SET current_job_status = %? WHERE table_id = %? AND current_job_status = %? AND current_job_id = %?: unsupported 1-th argument: running"] [id=dbaf9b58-8754-11ed-9b49-5405db7aad56]
[2022/12/29 09:43:25.108 +01:00] [INFO] [job_manager.go:417] ["scheduled ttl task"] [ttl-worker=manager] [jobID=dbaf9b58-8754-11ed-9b49-5405db7aad56] [table=t1]
[2022/12/29 09:43:25.109 +01:00] [INFO] [job_manager.go:292] ["scan task finished"] [ttl-worker=manager] [tableID=80] [jobID=dbaf9b58-8754-11ed-9b49-5405db7aad56]
[2022/12/29 09:43:35.106 +01:00] [INFO] [job_manager.go:334] ["job has finished"] [ttl-worker=manager] [jobID=dbaf9b58-8754-11ed-9b49-5405db7aad56] [statistics="Total Rows: 1, Success Rows: 1, Error Rows: 0"]
```
```

--- a/ttl/ttlworker/BUILD.bazel
+++ b/ttl/ttlworker/BUILD.bazel
@@ -41,6 +41,7 @@ go_test(
     name = "ttlworker_test",
     srcs = [
         "del_test.go",
+        "job_integration_test.go",
         "job_manager_integration_test.go",
         "job_manager_test.go",
         "job_test.go",

--- a/ttl/ttlworker/job.go
+++ b/ttl/ttlworker/job.go
@@ -47,7 +47,7 @@ const finishJobTemplate = `UPDATE mysql.tidb_ttl_table_status
 const updateJobStateTemplate = "UPDATE mysql.tidb_ttl_table_status SET current_job_state = %? WHERE table_id = %? AND current_job_id = %? AND current_job_owner_id = %?"
 
 func updateJobCurrentStatusSQL(tableID int64, oldStatus cache.JobStatus, newStatus cache.JobStatus, jobID string) (string, []interface{}) {
-	return updateJobCurrentStatusTemplate, []interface{}{newStatus, tableID, oldStatus, jobID}
+	return updateJobCurrentStatusTemplate, []interface{}{string(newStatus), tableID, string(oldStatus), jobID}
 }
 
 func finishJobSQL(tableID int64, finishTime time.Time, summary string, jobID string) (string, []interface{}) {

--- a/ttl/ttlworker/job_integration_test.go
+++ b/ttl/ttlworker/job_integration_test.go
@@ -1,0 +1,41 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ttlworker_test
+
+import (
+	"context"
+	"testing"
+
+	dbsession "github.com/pingcap/tidb/session"
+	"github.com/pingcap/tidb/testkit"
+	"github.com/pingcap/tidb/ttl/cache"
+	"github.com/pingcap/tidb/ttl/session"
+	"github.com/pingcap/tidb/ttl/ttlworker"
+	"github.com/stretchr/testify/require"
+)
+
+func TestChangeStatus(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+
+	dbSession, err := dbsession.CreateSession4Test(store)
+	require.NoError(t, err)
+	se := session.NewSession(dbSession, dbSession, nil)
+
+	job := ttlworker.NewTTLJob(&cache.PhysicalTable{ID: 0}, "0", cache.JobStatusWaiting)
+	tk.MustExec("insert into mysql.tidb_ttl_table_status(table_id,current_job_id,current_job_status) VALUES(0, '0', 'waiting')")
+	require.NoError(t, job.ChangeStatus(context.Background(), se, cache.JobStatusRunning))
+	tk.MustQuery("select current_job_status from mysql.tidb_ttl_table_status").Check(testkit.Rows("running"))
+}

--- a/ttl/ttlworker/job_test.go
+++ b/ttl/ttlworker/job_test.go
@@ -15,11 +15,26 @@
 package ttlworker
 
 import (
+	"context"
 	"testing"
 
 	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/ttl/cache"
+	"github.com/pingcap/tidb/ttl/session"
 	"github.com/stretchr/testify/assert"
 )
+
+func NewTTLJob(tbl *cache.PhysicalTable, id string, status cache.JobStatus) *ttlJob {
+	return &ttlJob{
+		tbl:    tbl,
+		id:     id,
+		status: status,
+	}
+}
+
+func (j *ttlJob) ChangeStatus(ctx context.Context, se session.Session, status cache.JobStatus) error {
+	return j.changeStatus(ctx, se, status)
+}
 
 func TestIterScanTask(t *testing.T) {
 	tbl := newMockTTLTbl(t, "t1")

